censys/asm/client.py

```diff
@@ -0,0 +1,146 @@
+"""
+Base for interacting with the Censys Search API.
+"""
+# pylint: disable=too-many-arguments
+import os
+
+from math import inf
+from typing import Type, Dict, Generator, Optional
+from requests.models import Response
+from censys.exceptions import *
+
+from censys.base import CensysAPIBase
+from censys.asm.seeds import Seeds
+from censys.asm.assets import Assets
+from censys.asm.events import Events
+from censys.config import get_config, DEFAULT
+
+
+class CensysAsmAPI(CensysAPIBase):
+    """
+    This is the base class for ASM's Seeds, Assets, and Events classes
+    """
+
+    DEFAULT_URL: str = "https://app.censys.io/api/v1"
+    """Default ASM API base URL."""
+    EXCEPTIONS: Dict[int, Type[CensysAsmException]] = {
+        10000: CensysMissingApiKeyException,
+        10008: CensysInvalidRequestException,
+        10002: CensysInvalidAuthTokenException,
+        10001: CensysInvalidAPIKeyException,
+        10039: CensysTooManyRequestsException,
+        10029: CensysAppDownForMaintenanceException,
+        10007: CensysInvalidSeedDataException,
+        10017: CensysAssociatedAssetsThresholdWarningException,
+        10016: CensysTooManyInputNodesException,
+        10014: CensysSeedNotFoundException,
+        10015: CensysNotASeedException,
+        10013: CensysNeedConfirmationToRemoveParentSeedsException,
+        10012: CensysCannotRemoveNonExistentSeedsException,
+        10011: CensysCannotRemoveNonSeedsException,
+        10038: CensysInvalidSeedTypeException,
+        10040: CensysInvalidLogbookCursorException,
+        10051: CensysPageNumberOutOfRangeException,
+        10050: CensysInvalidPageSizeException,
+        10018: CensysHostNotFoundException,
+        10021: CensysInvalidIPv4AddressException,
+        10054: CensysInvalidCommentException,
+        10055: CensysCommentNotFoundException,
+        10037: CensysInvalidColorException,
+        10025: CensysTagHasTrailingOrLeadingWhitespaceException,
+        10026: CensysTagIsEmptyStringException,
+        10027: CensysTagLabelsDifferOnlyInCasingException,
+        10028: CensysTagLabelTooLongException,
+        10034: CensysTagColorTooLongException,
+        10035: CensysCannotCreateTagWithNewColorException,
+        10036: CensysTagColorHasTrailingOrLeadingWhitespaceException,
+        10020: CensysCertificateNotFoundException,
+        10019: CensysDomainNotFoundException,
+    }
+    """Map of status code to Exception."""
+
+    def __init__(
+        self, api_key: Optional[str] = None, url: Optional[str] = DEFAULT_URL, **kwargs
+    ):
+        CensysAPIBase.__init__(self, url, **kwargs)
+
+        # Gets config file
+        config = get_config()
+
+        # Try to get credentials
+        self._api_key = (
+            api_key
+            or os.getenv("CENSYS_ASM_API_KEY")
+            or config.get(DEFAULT, "asm_api_key")
+        )
+
+        if not self._api_key:
+            raise CensysAsmException("No API key configured.")
```

Comment:
Getting this exception, I believe this is because this is the child class of `CensysAPIException` which requires both a status code and message. I would suggest using `CensysMissingApiKeyException` that is a child of `CensysException` or possibly making a new one.

```python
TypeError: __init__() missing 1 required positional argument: 'message'
```